national_rail/transform.py

```diff
@@ -0,0 +1,163 @@
+""" Module for transforming data received from the National Rail API into a 
+    useful format. """
+
+import logging
+
+from datetime import datetime, timedelta
+import xml.etree.ElementTree as ET
+from bs4 import BeautifulSoup
+
+
+def read_data_from_file(filename: str) -> str:
+    """ Opens file, reads and returns its content as a String. """
+
+    with open(filename, "r", encoding="utf-8") as file:
+        f = file.read()
+    return f
+
+
+def load_tree_root(national_rail_xml: str) -> ET.Element:
+    """ Returns root node of tree (from data.xml) """
+
+    tree = ET.ElementTree(ET.fromstring(national_rail_xml))
+    return tree.getroot()
+
+
+def reverse_tree(root: ET.Element) -> ET.Element:
+    """ Creates a new tree containing the root and its elements in reverse, to allow
+        for traversing the tree in reverse. Returns the new tree by its root. """
+
+    reversed_elements = list(reversed(list(root.iter())))
+    new_root = ET.Element("root")
+
+    for element in reversed_elements:
+        new_root.append(element)
+
+    return new_root
```

Comment:
```suggestion
    reversed_elements = [root.iter()][::-1]
    new_root = ET.Element("root")

    new_root.append(element for element in reversed_elements)

    return new_root
```

Reply:
I feel this can be more efficient

Reply:
Thanks, I've improved the code based on your suggestions. The append line didn't work as expected so I kept the original code.